torch/distributed/launch.py

```diff
@@ -0,0 +1,166 @@
+        --nnodes=2 --node_rank=1 --master_addr="192.168.1.1"
+        --master_port=1234 YOUR_TRAINING_SCRIPT.py (--arg1 --arg2 --arg3 and
+        all other arguments of your training script)``
+
+3. To look up what optional arguments this module offers:
+
+        >>> python -m torch.distributed.launch --help
+
+**Important Notices:**
+
+1. This utilty and multi-process distributed (single-node or
+multi-node) GPU training currently only achieves the best performance using
+the NCCL distributed backend. Thus NCCL backend is the recommended backend to
+use for GPU training.
+
+2. In your training program, you are supposed to parse the command-line
+argument: ``--device=DEVICE_TO_RUN``, (which will be provided with this module)
+and ensure that your code only runs on this device in your training program by:
+
+
+    >>> torch.cuda.set_device(arg.device)  # before your code runs
+
+    or
+
+    >>> with torch.cuda.device(arg.device):
+    >>>    # your code to run
+
+3. In your training program, you are supposed to call the following function
+at the beginning to start the distributed backend. You need to make sure that
+the init_method uses ``env://``, which is the only supported ``init_method``
+by this module:
+
+    ``torch.distributed.init_process_group(backend='YOUR BACKEND',
+    init_method='env://')``
+
+4. In your training program, you can either use regular distributed functions
+or use DistributedDataParallel module. If you would like to use
+DistributedDataParallel module, here is how to configure it.
+
+    ``model = torch.nn.parallel.DistributedDataParallel(model,
+    device_ids=[arg.device], output_device=arg.device)``
+
+Please ensure that ``device_ids`` argument is set to be the only GPU device id
+that your code will be operating on. In other words, the ``device_ids`` needs
+to be ``[args.device]``, and ``output_device`` needs to be ``args.devices`` in
+order to use this utility
+
+"""
+
+
+import subprocess
+import os
+import socket
+from argparse import ArgumentParser, REMAINDER
+
+import torch
+
+
+def parse_args():
+    """
+    Helper function parsing the command line options
+    @retval ArgumentParser
+    """
+    parser = ArgumentParser(description="PyTorch distributed training launch "
+                                        "helper utilty that will spawn up "
+                                        "multiple distributed processes")
+
+    # Optional arguments for the launch helper
+    parser.add_argument("--nnodes", type=int, default=1,
+                        help="The number of nodes to use for distributed "
+                             "training")
+    parser.add_argument("--node_rank", type=int, default=0,
+                        help="The rank of the node for multi-node distributed "
+                             "training")
+    parser.add_argument("--nproc_per_node", type=int, default=1,
+                        help="The number of processes to launch on each node, "
```

Comment:
I don't see nproc_per_node being reset to the number of GPUs on the system?

Reply:
@ngimel  your comment is addressed in the latest commit